### ISSUE TITLE
The difference between YOLO v9c-seg　and　gelan-c-seg？

### ISSUE THREAD

**Issue by Thompson21666** [WongKinYiu/yolov9#633]
１）The difference between YOLO v9c-seg　and　gelan-c-seg？
２）The difference between　gelan-c-seg　and　gelan-c-dseg？
３）In addition to the YOLO v9c-seg　and　gelan-c-seg, 　what are the segmentation models that have been made public?

![Image](https://github.com/user-attachments/assets/0fc0740b-dcab-4090-884d-b533c7e8f1f9)
４）I also found the following two, 　YOLO v9c-dseg　and　gelan-c-dseg,　why are they not published on the main page? Can it be used for semantic segmentation training?

![Image](https://github.com/user-attachments/assets/03f5b9a4-441b-42b2-9a5e-64baecc2d22b)

**Comment by fazrigading** [comment]
1. YOLOv9 using GELAN+PGI in its architecture, while GELAN models are only using GELAN without PGI. Read the paper [here](https://arxiv.org/pdf/2402.13616), page 10, figure 7. Description stated: 

> As for GELAN that does not use PGI, we found that it had divergence when detecting object boundaries, and it also produced unexpected responses in some background areas. This experiment confirms that PGI can indeed provide better gradients to update parameters and enable the feedforward stage of the main branch to retain more important features.

2. "seg" means the model uses Segment class in `models/yolo.py#L419` , which is for the "normal" YOLOv9 segmentation model: https://github.com/WongKinYiu/yolov9/blob/5b1ea9a8b3f0ffe4fe0e203ec6232d788bb3fcff/models/yolo.py#L419

Meanwhile "dseg" means the model uses DSegment class in `models/yolo.py#L442`, which is for the decoupled version, where detection and mask generation are more cleanly split:
https://github.com/WongKinYiu/yolov9/blob/5b1ea9a8b3f0ffe4fe0e203ec6232d788bb3fcff/models/yolo.py#L442

3.  Only 4 models, please check the releases page: https://github.com/WongKinYiu/yolov9/releases/tag/v0.1
4. The author(s) possibly did neither trained the YOLOv9-dseg or GELAN-dseg nor published the weights. However, you could use it for training to your own dataset. But, you need to train them first to COCO dataset by using the same hyperparams used by the author for YOLOv9-seg and GELAN-seg in order to produce the generalization for 80 objects, it would take a long time without using recommended GPUs. I recommend not to train a model from scratch without weights if you don't have that much resources.

**Comment by Thompson21666** [comment]
> 1. YOLOv9 using GELAN+PGI in its architecture, while GELAN models are only using GELAN without PGI. Read the paper [here](https://arxiv.org/pdf/2402.13616), page 10, figure 7. Description stated:
> 
> > As for GELAN that does not use PGI, we found that it had divergence when detecting object boundaries, and it also produced unexpected responses in some background areas. This experiment confirms that PGI can indeed provide better gradients to update parameters and enable the feedforward stage of the main branch to retain more important features.
> 
> 2. "seg" means the model uses Segment class in `models/yolo.py#L419` , which is for the "normal" YOLOv9 segmentation model: 
>      
>        
>          [yolov9/models/yolo.py](https://github.com/WongKinYiu/yolov9/blob/5b1ea9a8b3f0ffe4fe0e203ec6232d788bb3fcff/models/yolo.py#L419)
>        
>        
>             Line 419
>          in
>          [5b1ea9a](/WongKinYiu/yolov9/commit/5b1ea9a8b3f0ffe4fe0e203ec6232d788bb3fcff)
>        
>      
>      
>        
>    
>            
>              
>               class Segment(Detect):
> 
> Meanwhile "dseg" means the model uses DSegment class in `models/yolo.py#L442`, which is for the decoupled version, where detection and mask generation are more cleanly split:
> 
> [yolov9/models/yolo.py](https://github.com/WongKinYiu/yolov9/blob/5b1ea9a8b3f0ffe4fe0e203ec6232d788bb3fcff/models/yolo.py#L442)
> 
> Line 442 in [5b1ea9a](/WongKinYiu/yolov9/commit/5b1ea9a8b3f0ffe4fe0e203ec6232d788bb3fcff)
> 
>  class DSegment(DDetect): 
> 3. Only 4 models, please check the releases page: https://github.com/WongKinYiu/yolov9/releases/tag/v0.1
> 4. The author(s) possibly did neither trained the YOLOv9-dseg or GELAN-dseg nor published the weights. However, you could use it for training to your own dataset. But, you need to train them first to COCO dataset by using the same hyperparams used by the author for YOLOv9-seg and GELAN-seg in order to produce the generalization for 80 objects, it would take a long time without using recommended GPUs. I recommend not to train a model from scratch without weights if you don't have that much resources.

Many thanks.

**Comment by fazrigading** [comment]
> Many thanks.

You are welcome.

**Comment by fazrigading** [comment]
@Thompson21666 You can close this issue.
